modules/lib/nushell.nix

```diff
@@ -0,0 +1,63 @@
+{ lib }: rec {
+  mkNushellInline = expr: lib.setType "nushell-inline" { inherit expr; };
+
+  toNushell = { indent ? "", multiline ? true, asBindings ? false }@args:
```

Comment:
I guess `multiline`, `indent` (and vars `outroSpace`, `introSpace`) is purely for aesthetic reasons?

`asBindings` isn't used anywhere as well (or rather is false everywhere), for what is this used, is something planned with this?

(No need for action, just questions)

Reply:
This is straight up yoinked from [the `nixpkgs` lua generator](https://github.com/NixOS/nixpkgs/blob/8637e879e425363b14aa23f3ec1f69e32c9f512f/lib/generators.nix#L704). I thought about removing some unused features it but they work so why not keep them.

`multiline` et al. are just aesthetic, but I like that it's the default to avoid having incredibly long lines in the config files.

`asBindings` could be used by consumers of the lib but it is unused by the module as well.